src/components/NFTOriginVerifier.tsx

```diff
@@ -245,27 +316,58 @@ export default function NFTOriginVerifier() {
               </MenuItem>
             ))}
           </TextField>
-          {isEVMChain(lookupChain) ? (
-            <KeyAndBalance chainId={lookupChain} />
-          ) : null}
-          <TextField
-            fullWidth
-            variant="outlined"
-            margin="normal"
-            label="Paste an address"
-            value={lookupAsset}
-            onChange={handleAssetChange}
-          />
-          {isEVMChain(lookupChain) ? (
-            <TextField
-              fullWidth
-              variant="outlined"
-              margin="normal"
-              label="Paste a tokenId"
-              value={lookupTokenId}
-              onChange={handleTokenIdChange}
-            />
-          ) : null}
+          {lookupChain === CHAIN_ID_APTOS ? (
+            <>
+              <TextField
+                fullWidth
+                variant="outlined"
+                margin="normal"
+                label="Paste creator address"
+                value={lookupCreatorAddress}
+                onChange={handleCreatorAddressChange}
+              />
+              <TextField
+                fullWidth
+                variant="outlined"
+                margin="normal"
+                label="Paste collection name"
+                value={lookupCollectionName}
+                onChange={handleCollectionNameChange}
+              />
+              <TextField
+                fullWidth
+                variant="outlined"
+                margin="normal"
+                label="Paste token name"
+                value={lookupTokenName}
+                onChange={handleTokenNameChange}
+              />
+            </>
+          ) : (
+            <>
+              {isEVMChain(lookupChain) ? (
+                <KeyAndBalance chainId={lookupChain} />
+              ) : null}
```

Comment:
We could use `short-circiut` here, so only when `isEVMChain(lookupChain) ` is truth will display `KeyAndBalance`

```suggestion
              {isEVMChain(lookupChain) &&
                <KeyAndBalance chainId={lookupChain} />}
```

Reply:
fixed